operator/src/main/java/oracle/kubernetes/operator/WaitForReadyStep.java

```diff
@@ -197,17 +209,32 @@ public String getName() {
     return new DefaultResponseStep<>(null) {
       @Override
       public NextAction onSuccess(Packet packet, CallResponse<T> callResponse) {
+        DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
         if ((callResponse != null) && (callResponse.getResult() instanceof V1Pod)) {
           V1Pod pod = (V1Pod) callResponse.getResult();
-          Optional.ofNullable(packet.getSpi(DomainPresenceInfo.class))
+          Optional.ofNullable(info)
                   .ifPresent(i -> i.setServerPodFromEvent(getPodLabel(pod, LabelConstants.SERVERNAME_LABEL), pod));
         }
-        if (isReady(callResponse.getResult())) {
+        if (isReady(callResponse.getResult()) || callback.didResume.get()) {
```

Comment:
I don't understand why we need to check callback.didResume here. How would the operator get here again if the callback has been resumed already?

Reply:
The check callback.didResume is to avoid the race condition. We do the periodic listing of the introspector job to check if it's ready and also dispatch the callback when the watch event is received. In this scenario, the watch event notifications are flowing. After the intro job completed watch notification is received, the callback is removed and the fiber is resumed. After this, the intro job gets deleted but the child fiber that's listing the job periodically to check for readiness doesn't see the job as ready since it got deleted. Hence the child fiber never finishes and times out. The above check will terminate the child fiber after intro job has been deleted.

Reply:
If I understand it correctly, check for didResume (didResumeFiber now) is only needed for the job case. If so, we need to remove the check in the pod code path,

Reply:
The `didResumeFiber` will return true only when the fiber is resumed and the resource should be ready at that time. I think there's no harm in keeping the check as that should avoid any potential race condition in pod case as well.